sonar-plugin-server/src/main/java/jetbrains/buildserver/sonarplugin/SonarQubeSecureParametersProcessor.java

```diff
@@ -0,0 +1,84 @@
+package jetbrains.buildserver.sonarplugin;
+
+import com.intellij.openapi.diagnostic.Logger;
+import jetbrains.buildServer.serverSide.*;
+import jetbrains.buildServer.serverSide.executors.ExecutorServices;
+import jetbrains.buildServer.util.EventDispatcher;
+import jetbrains.buildserver.sonarplugin.manager.MigratingSQSManager;
+import jetbrains.buildserver.sonarplugin.manager.SQSInfo;
+import jetbrains.buildserver.sonarplugin.manager.SQSManager;
+import jetbrains.buildserver.sonarplugin.manager.projectfeatures.SQSInfoImpl;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SonarQubeSecureParametersProcessor extends BuildServerAdapter {
+    private final static String PASSWORD = "password";
+    private final static String TOKEN = "token";
+    private final static String JDBC_PASSWORD = "jdbcPassword";
+
+    @NotNull
+    private final ProjectManager myProjectManager;
+    @NotNull
+    private final MigratingSQSManager myMigratingSQSManager;
+    @NotNull
+    private final ExecutorServices myExecutorServices;
+    @NotNull
+    private final ConfigActionFactory myConfigActionFactory;
+    @NotNull
+    private final static Logger LOG = Logger.getInstance(SonarQubeSecureParametersProcessor.class);
+
+
+    public SonarQubeSecureParametersProcessor(@NotNull MigratingSQSManager migratingSQSManager,
+                                              @NotNull ProjectManager myProjectManager,
+                                              @NotNull EventDispatcher<BuildServerListener> dispatcher,
+                                              @NotNull ExecutorServices executorServices,
+                                              @NotNull ConfigActionFactory myConfigActionFactory) {
+        this.myProjectManager = myProjectManager;
+        this.myMigratingSQSManager = migratingSQSManager;
+        this.myExecutorServices = executorServices;
+        this.myConfigActionFactory = myConfigActionFactory;
+        dispatcher.addListener(this);
+    }
+
+    @Override
+    public void serverStartup() {
+        myExecutorServices.getLowPriorityExecutorService().submit(() -> {
```

Comment:
In rare cases submit can throw a RejectedExecutionException, so let's add a try catch for it too.